discovery-provider/src/tasks/index_metrics.py

```diff
@@ -126,7 +136,144 @@ def refresh_metrics_matviews(db):
         session.execute('REFRESH MATERIALIZED VIEW app_name_metrics_all_time')
         logger.info('index_metrics.py | refreshed metrics matviews')
 
-######## CELERY TASK ########
+def get_all_other_nodes():
```

Comment:
we're definitely going to forget the algorithm here so maybe a longer docstring explaining why we do certain things would be pay dividends in the future. maybe you already planned to do this :)